File: leetcode/1323. 6 和 9 组成的最大数字.go
func maximum69Number(num int) int {
	numstr := strconv.Itoa(num)
	result := ""
	found := false
	for _, ch := range numstr {
		if !found && ch == '6' {
			result += "9"
			found = true
		} else {
			result += string(ch)
		}
	}
	res, _ := strconv.Atoi(result)
	return res
}

File: leetcode/1323. 6 和 9 组成的最大数字.rs
impl Solution {
    pub fn maximum69_number(num: i32) -> i32 {
        let mut num_str = num.to_string();
        
        let mut chars: Vec<char> = num_str.chars().collect();

        for i in 0..chars.len() {
            if chars[i] == '6' {
                chars[i] = '9';
                break; // 只替换第一个6
            }
        }
        
        let modified_str: String = chars.into_iter().collect();
        
        modified_str.parse().expect("Failed to parse number")
    }
}